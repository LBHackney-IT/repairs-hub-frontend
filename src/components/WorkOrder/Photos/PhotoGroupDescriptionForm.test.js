import { render } from '@testing-library/react'
import {
  UpdateDescriptionButton,
  UpdateDescriptionForm,
} from './PhotoGroupDescriptionForm'

describe('UpdateDescriptionButton component', () => {
  it('renders UpdateDescriptionButton component with no description', () => {
    const { asFragment } = render(
      <UpdateDescriptionButton
        description={''}
        showForm={() => {}}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders UpdateDescriptionButton component with description', () => {
    const { asFragment } = render(
      <UpdateDescriptionButton
        description={'Some description'}
        showForm={() => {}}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders UpdateDescriptUpdateDescriptionFormonButton component with no description', () => {
    const { asFragment } = render(
      <UpdateDescriptionForm
        description={''}
        onSubmit={() => {}}
        fileGroupId={1}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders UpdateDescriptUpdateDescriptionFormonButton component with description', () => {
    const { asFragment } = render(
      <UpdateDescriptionForm
        description={'Some description'}
        onSubmit={() => {}}
        fileGroupId={1}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
