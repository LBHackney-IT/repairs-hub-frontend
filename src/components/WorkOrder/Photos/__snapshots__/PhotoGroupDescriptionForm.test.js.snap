// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateDescriptionButton component renders UpdateDescriptUpdateDescriptionFormonButton component with description 1`] = `
<DocumentFragment>
  <div
    class="govuk-grid-row"
  >
    <div
      class="govuk-grid-column-two-thirds"
    >
      <form
        id="description-form"
        role="form"
      >
        <div
          class="govuk-form-group lbh-form-group"
          id="description-form-group"
        >
          <label
            class="govuk-label lbh-label"
            for="description"
          >
            Description
          </label>
          <textarea
            aria-describedby="description-hint description-error"
            class="govuk-textarea lbh-textarea"
            data-testid="description"
            id="description"
            name="description"
            rows="4"
          >
            Some description
          </textarea>
        </div>
        <input
          id="fileGroupId"
          label="fileGroupId"
          name="fileGroupId"
          type="hidden"
          value="1"
        />
        <div
          class="govuk-form-group lbh-form-group"
        >
          <button
            class="govuk-button lbh-button"
            data-module="govuk-button"
            type="submit"
          >
            Save description
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;

exports[`UpdateDescriptionButton component renders UpdateDescriptUpdateDescriptionFormonButton component with no description 1`] = `
<DocumentFragment>
  <div
    class="govuk-grid-row"
  >
    <div
      class="govuk-grid-column-two-thirds"
    >
      <form
        id="description-form"
        role="form"
      >
        <div
          class="govuk-form-group lbh-form-group"
          id="description-form-group"
        >
          <label
            class="govuk-label lbh-label"
            for="description"
          >
            Description
          </label>
          <textarea
            aria-describedby="description-hint description-error"
            class="govuk-textarea lbh-textarea"
            data-testid="description"
            id="description"
            name="description"
            rows="4"
          />
        </div>
        <input
          id="fileGroupId"
          label="fileGroupId"
          name="fileGroupId"
          type="hidden"
          value="1"
        />
        <div
          class="govuk-form-group lbh-form-group"
        >
          <button
            class="govuk-button lbh-button"
            data-module="govuk-button"
            type="submit"
          >
            Add description
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;

exports[`UpdateDescriptionButton component renders UpdateDescriptionButton component with description 1`] = `
<DocumentFragment>
  <div
    class="display-inline"
  >
    <p
      style="color: rgb(102, 102, 102); margin-top: 15px;"
    >
      Some description
    </p>
    <button
      class="lbh-link updateDescriptionButton"
      type="button"
    >
      Edit description
    </button>
  </div>
</DocumentFragment>
`;

exports[`UpdateDescriptionButton component renders UpdateDescriptionButton component with no description 1`] = `
<DocumentFragment>
  <div
    class="display-inline"
  >
    <button
      class="lbh-link updateDescriptionButton"
      type="button"
    >
      Add description
    </button>
  </div>
</DocumentFragment>
`;
